eldritch/test_specials.py

```diff
@@ -98,3 +105,137 @@ def testCursedWhileBlessed(self):
+      self.assertEqual(roll.call_count, 1)
+
+    with mock_randint(1) as roll:
+      self.advance_turn(3, "upkeep")
+      interest = self.resolve_to_choice(events.SpendChoice)
+      with self.assertRaises(game.InvalidMove):
+        interest.resolve(self.state, "No")
+      self.spend("dollars", 1, interest)
+      interest.resolve(self.state, "Yes")
+      choice = self.resolve_to_choice(events.SliderInput)
+      choice.resolve(self.state, "done", "done")
+      payoff = self.resolve_to_usable(0, "Bank Loan0")
+      # Probably shouldn't be usable if you're too poor
+      self.state.done_using[0] = True
+      self.resolve_until_done()
+      self.assertEqual(roll.call_count, 1)
+
+    self.char.dollars = 10
+
+    with mock_randint(5) as roll:
+      self.advance_turn(4, "upkeep")
+      choice = self.resolve_to_choice(events.SliderInput)
+      choice.resolve(self.state, "done", "done")
+      payoff = self.resolve_to_usable(0, "Bank Loan0")
+      self.state.event_stack.append(payoff)
+      payoff_choice = self.resolve_to_choice(events.SpendChoice)
+      self.spend("dollars", 10, payoff_choice)
+      payoff_choice.resolve(self.state, "Yes")
+      self.resolve_until_done()
+      self.assertEqual(roll.call_count, 1)
+    self.assertEqual(self.char.dollars, 0)
+    self.assertNotIn("Bank Loan", [p.name for p in self.char.possessions])
+    self.assertFalse(self.char.possessions)
+
+  def defaultSequence(self) -> events.ItemLossChoice:
+    self.advance_turn(0, "encounter")
+    self.resolve_until_done()
+    self.state.event_stack.append(events.TakeBankLoan(self.char))
+    self.resolve_until_done()
+    self.assertEqual(self.char.dollars, 10)
+    self.char.dollars = 0
+    with mock_randint(3) as roll:
+      self.advance_turn(1, "upkeep")
+      choice = self.resolve_to_choice(events.SliderInput)
+      choice.resolve(self.state, "done", "done")
+      self.resolve_to_usable(0, "Bank Loan0")
+      self.state.done_using[0] = True
+      self.resolve_until_done()
+      self.advance_turn(2, "upkeep")
+      interest = self.resolve_to_choice(events.SpendChoice)
+      interest.resolve(self.state, "No")
+      self.assertEqual(roll.call_count, 1)
+      return self.resolve_to_choice(events.ItemLossChoice)
+
+  def testDefault(self):
+    self.char.possessions = [items.Whiskey(0), items.Wither(0), items.MagicPowder(0)]
+    loss_choice = self.defaultSequence()
+    loss_choice.resolve(self.state, "Whiskey0")
+    loss_choice.resolve(self.state, "Wither0")
+    with self.assertRaises(eldritch.eldritch.InvalidMove):
+      loss_choice.resolve(self.state, "done")
+    loss_choice.resolve(self.state, "Magic Powder0")
+    loss_choice.resolve(self.state, "done")
+    self.resolve_to_choice(events.SliderInput)
+    self.assertListEqual([p.name for p in self.char.possessions], ["Bad Credit"])
+
+  def testDefaultWithDerringer(self):
+    self.char.possessions = [items.Derringer18(0), items.Wither(0), items.MagicPowder(0)]
```

Comment:
Can you also give them a retainer to make sure they don't have to discard any specials?

Reply:
A Retainer makes it complicated, since the retainer gives them money and prevents the default. I gave a Lodge Membership instead.